src/views/issuance/components/zap/state/atoms.ts

```diff
@@ -0,0 +1,378 @@
+import {
+  base,
+  contracts,
+  entities,
+  searcher,
+} from '@reserve-protocol/token-zapper'
+import {
+  PERMIT2_ADDRESS,
+  PermitTransferFrom,
+  PermitTransferFromData,
+  SignatureTransfer,
+} from '@uniswap/permit2-sdk'
+import { ethers } from 'ethers'
+import { atom, Getter } from 'jotai'
+import { loadable } from 'jotai/utils'
+import { rTokenAtom, walletAtom } from 'state/atoms'
+import { tokenBalancesStore } from 'state/TokenBalancesUpdater'
+
+import atomWithDebounce from 'utils/atoms/atomWithDebounce'
+import {
+  atomWithOnWrite,
+  onlyNonNullAtom,
+  simplifyLoadable,
+} from 'utils/atoms/utils'
+
+import {
+  resolvedZapState,
+  supportsPermit2Signatures,
+  zappableTokens,
+} from './zapper'
+
+/**
+ * I've tried to keep react effects to a minimum so most async code is triggered via some signal
```

Comment:
I see the reasoning, but I kind of like leaning more towards abstracting this checks into a hook, `useZap` or something that returns this state, not something to change right now